sound/soc/sof/core.c

```diff
@@ -350,7 +350,9 @@ static int sof_init_environment(struct snd_sof_dev *sdev)
 	}
 
 	ret = sof_select_ipc_and_paths(sdev);
-	if (!ret && plat_data->ipc_type != base_profile->ipc_type) {
+	if (ret) {
+		goto err_machine_check;
+	} else if (plat_data->ipc_type != base_profile->ipc_type) {
```

Comment:
no longer a fixup, the initial patch was applied by Mark Brown

Reply:
oh, OK, then I will reword on Monday